docs/workflows/common-workflow-steps/custom-script-from-git.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Custom Script from Git
+description: Use Script steps for additional functionalities in your builds.
+tags: [custom scripts, build, test, workflow, step]
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import Screenshot from '@site/src/components/Screenshot';
+import SensitiveVariablesDanger from '@site/docs/\_sensitive-variables-danger.mdx';
+
+# Custom Script from Git
+
+You can use **Custom Script from Git** to clone and run your own scripts directly from a Git repository as part of your Appcircle build. This step supports authenticated cloning (via username and PAT).
+
+Before execution, the step will clone (or reuse) your repository, check out the specified branch, and execute the script based on its file extension (postfix).
+
+:::caution
+
+Keep in mind that the script file has to be under the repository root folder in order to work with the component.
+
+:::
+
+:::info
+
+The step inspects your script file’s postfix (its extension—.sh, .py, .rb, .pl, .js, .java) to choose the execution path. It must match exactly.
```

Comment:
Please use backticks `` to format technical terms and code elements for better readability and clarity.